server/src/main/java/org/elasticsearch/action/admin/cluster/repositories/verify/VerifyRepositoryRestResponse.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.action.admin.cluster.repositories.verify;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class VerifyRepositoryRestResponse implements ToXContentObject {
```

Comment:
I agree that we need a different response. In this case I wonder if we should change the transport client response too and align it with REST. I do not see also why transport client returns the cluster name. In that case, we would introduce the needed changes to the existing response and break the transport client. Otherwise, I think the new response should be placed among the high-level client classes. @nik9000 which way would you go?

Reply:
Yeah, I'd be tempted to modify the response that we use in the server and transport client to look more like the rest response rather than make a whole new response. When we talked we weren't clear on whether or not we all felt like that was ok to do though. I think it is fine in cases where we don't **need** the extra data. But if we're uncomfortable making that change I can understand it. If we don't want to change the response then, yeah, I'd make a new response object and pitch it into the high level rest client.

Reply:
I think we need to go case by case and this is one of the cases where it should not make a difference to return something at transport that resembles more what we return at REST, so I tend to think that this API is a good example of where we should change the existing reponse, especially given how small the response is. It will be a breaking change for transport client users though.

Reply:
Yeah. We've traditionally been fairly ok with breaking transport client users in minor releases if we have a good reason. To me this is a good reason.

Reply:
oh that is a good point. the 6.x class will still read and write a discovery node depending on the version but it can live without it? When you look at how much stuff we serialize over the wire, you do see how this change is useful at transport too (unless we need all this info somewhere). Would that work?

Reply:
I am going to remove the `DiscoveryNode` from the 7.0 class, read/write `NodeView `and the `response.nodes()` will return a `List<NodeView>`. in 6.x i will still read/write `DiscoveryNode` and `response.nodes()` will convert from `DiscoveryNode[]` to `List<NodeView>` on the fly, so there is only 1 backing object in each version of the code. Sound valid?

Reply:
++, the only thing is I would even go with a Map<String,String> if that works. not sure what you and Nik think about that.

Reply:
ok I chatted with @nik9000 and @jasontedor and I will do 2 PRs. The first one (this one) will keep the backing object a `DiscoveryNode[]`, but will read/write the `NodeView` class to >=6.4 nodes, and write the `DiscoveryNode` otherwise. Then once thats backported, ill come back in on the master branch and fix it so that it knows zero about the `DiscoveryNode`, and instead just reads/writes and stores a `List<NodeView>`

Reply:
linking #30762 as I will make sure these changes are done independent of this PR.